soda-data-quality-checks/README.md

```diff
@@ -0,0 +1,45 @@
+# Demo: Soda data quality checks with LocalStack Snowflake 
+
+This project illustrates how to use the [Soda framework](https://www.soda.io/) to run data quality checks against Snowflake tables, entirely on your local machine.
+
+The code is based on the Snowflake Quickstart Guide on [Data Quality Testing with Soda](https://quickstarts.snowflake.com/guide/soda).
+
+## Prerequisites
+
+- [`localstack` CLI](https://docs.localstack.cloud/getting-started/installation/#localstack-cli) with [`LOCALSTACK_AUTH_TOKEN`](https://docs.localstack.cloud/getting-started/auth-token/) environment variable set
+- [`awslocal` CLI](https://docs.localstack.cloud/user-guide/integrations/aws-cli/#localstack-aws-cli-awslocal) 
+- [LocalStack Snowflake emulator](https://snowflake.localstack.cloud/getting-started/installation/)
+- [`snow` CLI](https://snowflake.localstack.cloud/user-guide/integrations/snow-cli/) with `local` connection profile pointing to LocalStack Snowflake
+
+## Instructions
+
+### Start LocalStack
+
+Start the LocalStack Snowflake emulator using the following command:
+
+```bash
+DOCKER_FLAGS='-e SF_LOG=trace' \
+  IMAGE_NAME=localstack/snowflake \
+  DEBUG=1 \
+  localstack start
+```
+
+### Initialize the data tables
+
+The sample application provides Makefile targets to simplify the setup process. 
+
+Run the following command to initialize the environment and seed test data into local Snowflake:
+```
+make init  
```

Comment:
I think make install needs to run, I ran `make init` without `make install` and I got back error `soda not found`

Reply:
After running `make install` and starting LocalStack, begin running `make scan`, these are my current results:

```text
soda test-connection -d local_snowflake -c configuration.yml
[10:26:10] Soda Core 3.3.10
[10:26:12] Could not connect to data source "local_snowflake": 250001: Could not connect to Snowflake backend after 2 attempt(s).Aborting
  | 250001: Could not connect to Snowflake backend after 2 attempt(s).Aborting
[10:26:14] Could not connect to data source "local_snowflake": 250001: Could not connect to Snowflake backend after 2 attempt(s).Aborting
  | 250001: Could not connect to Snowflake backend after 2 attempt(s).Aborting
Unable to connect to 'local_snowflake'.
Traceback (most recent call last):
  File "/home/yingw787/.pyenv/versions/3.11.6/bin/soda", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/soda/telemetry/soda_tracer.py", line 78, in wrapper
    raise e
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/soda/telemetry/soda_tracer.py", line 71, in wrapper
    result = fn(*original_args, **original_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yingw787/.pyenv/versions/3.11.6/lib/python3.11/site-packages/soda/cli/cli.py", line 602, in test_connection
    has_valid_connection = ds.has_valid_connection()
                           ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'has_valid_connection'
```

Reply:
Ran a gut check with `curl -d '{}' snowflake.localhost.localstack.cloud:4566/session` and confirmed LocalStack is running correctly. Not quite sure if this is reproducible?